packages/nx-monorepo/scripts/pnpm/link-bundled-transitive-deps.ts

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env ts-node
+/*! Copyright [Amazon.com](http://amazon.com/), Inc. or its affiliates. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0 */
+import * as path from "node:path";
+import * as fs from "fs-extra";
+import * as SemVer from "semver";
+import { buildDependenciesHierarchy, PackageNode } from "@pnpm/reviewing.dependencies-hierarchy";
+
+async function linkBundledTransitiveDeps(workspaceDir: string, pkgFolder: string) {
+  const pkgDir = path.join(workspaceDir, pkgFolder);
+  const pkgJson = require(path.join(pkgDir, "package.json"));
+  const bundledDeps: string[] = pkgJson.bundledDependencies || [];
+  if (bundledDeps.length < 1) {
+    // No bundled deps
+    return;
+  }
+
+  const dependencyHierarchy = (await buildDependenciesHierarchy([pkgDir], {
+    depth: Number.MAX_SAFE_INTEGER,
+    lockfileDir: workspaceDir,
+    include: {
+      optionalDependencies: false,
+      dependencies: true,
+      devDependencies: false,
+    },
+  }))[pkgDir];
+
+  const transitiveDeps: Record<string, PackageNode> = {};
+
+  function visit(_deps?: PackageNode[]) {
+    if (_deps == null || !_deps.length) {
```

Comment:
deps is optional so we also need to check for `!_deps` before checking `!_deps.length`

Reply:
_deps == null is not the same as _deps === undefined right?